src/Cooked/MockChain/Direct.hs

```diff
@@ -189,11 +190,15 @@ instance (Monad m) => MonadBlockChainWithoutValidation (MockChainT m) where
   setParams params = do
     modify $ set mcstParamsL params
     modify $ over mcstLedgerStateL (Emulator.updateStateParams params)
-  currentSlot = gets (Emulator.getSlot . mcstLedgerState)
-  awaitSlot slot = do
-    cs <- currentSlot
-    when (slot > cs) $ modify' (over mcstLedgerStateL (Lens.set Emulator.elsSlotL (Ledger.toCardanoSlotNo slot)))
-    return $ max slot cs
+  waitNSlots n = do
+    cs <- gets (Emulator.getSlot . mcstLedgerState)
+    if
```

Comment:
I knew about guards in function definitions but I only discover now about this if + guards construction in do-notations. Convenient. Is this provided by a language extension or vanilla Haskell?

Reply:
this comes from the `MultiWayIf` language extension ! Pretty convenient to avoid nested `if then else` !